### ISSUE TITLE
[BUG] block in block error message should not show without repeat/if

### ISSUE THREAD

**Issue by Felienne** [hedyorg/hedy#3270]
**Describe the bug**
A block in a block error is given when we do not use a repeat.

**Paste the Hedy code & level**
Level 7
```
repeat 5 times
  print 'Hello folks'
    print 'This will be printed 5 times'
print 'We are done now'
```

**Add a screenshot (optional)**
![image](https://user-images.githubusercontent.com/1003685/189379822-0d1f77e9-9516-4e1c-bf54-c7e709d84979.png)

**Comment by Felienne** [comment]
Related but slightly different:

![image](https://user-images.githubusercontent.com/1003685/189380998-820bbda7-1e2f-4c97-96b9-9889f71d63de.png)

**Comment by Felienne** [comment]
Also for @boryanagoncharenko ?

**Comment by Felienne** [comment]
> @Felienne What errors should we display to the user in both cases?
> 
> In the first error, instead of `You are using a block in a block! That is awesome, but a block in a block is not unlocked yet! It will be unlocked in a later level` we could show `You used too many spaces in line 3. You used 4 spaces, which is too much. Start every new block with 2 spaces more than the line before.` However, this is also a bit misleading because the last sentence does talk about "every new block". Maybe we could just show `You used too many spaces in line 3. You used 4 spaces, which is too much.`?

Yes, I think `You used too many spaces in line 3. You used 4 spaces, which is too much.` would be a perfect error message if we can get that to work!
 
> The second error cannot be replicated anymore. 

Ah I guess @akseron fixed that already (or maybe I did it?) 

> Now we show the message `You used too many spaces in line 4. You used 2 spaces, which is too much. Start every new block with 2 spaces more than the line before.` This message seems correct but, again, I find the last sentence confusing because it directs the user to a 'new block'. Perhaps we could remove the last sentence for level 8?

Yeah I agree that final sentence can be removed in this case! But maybe not in all cases? I am not sure if there are cases where it should remain. What do you think?

**Comment by akseron** [comment]
> > @Felienne What errors should we display to the user in both cases?
> > In the first error, instead of `You are using a block in a block! That is awesome, but a block in a block is not unlocked yet! It will be unlocked in a later level` we could show `You used too many spaces in line 3. You used 4 spaces, which is too much. Start every new block with 2 spaces more than the line before.` However, this is also a bit misleading because the last sentence does talk about "every new block". Maybe we could just show `You used too many spaces in line 3. You used 4 spaces, which is too much.`?
> 
> Yes, I think `You used too many spaces in line 3. You used 4 spaces, which is too much.` would be a perfect error message if we can get that to work!
> 
> > The second error cannot be replicated anymore.
> 
> Ah I guess @akseron fixed that already (or maybe I did it?)
> 
> > Now we show the message `You used too many spaces in line 4. You used 2 spaces, which is too much. Start every new block with 2 spaces more than the line before.` This message seems correct but, again, I find the last sentence confusing because it directs the user to a 'new block'. Perhaps we could remove the last sentence for level 8?
> 
> Yeah I agree that final sentence can be removed in this case! But maybe not in all cases? I am not sure if there are cases where it should remain. What do you think?

Yes I remember looking at it, however it was already solved in  #4351 😄
